= About the GWAS Catalog

== About the GWAS Catalog

The Catalog was founded by the https://www.genome.gov[NHGRI] in 2008, in response to the rapid increase in the number of published genome-wide association studies (GWAS). These studies provide an unprecedented opportunity to investigate the impact of common variants on complex disease; however identifying published GWAS can be challenging, and the vast wealth of data contained within these publications is effectively inaccessible to researchers without systematic cataloguing and summarization of the observed associations. The GWAS Catalog provides a consistent, searchable, visualisable and freely available database of SNP-trait associations, which can be easily integrated with other resources, and is accessed by scientists, clinicians and other users worldwide.

Within the Catalog, all link:methods[eligible published] GWA studies are identified by literature search and assessed by our curators, who then extract the reported trait, significant SNP-trait associations, and sample metadata. We aim to curate eligible studies within 1-2 months of publication, dependent on the availability of literature, and the data is released on a weekly cycle. We have also accepted submissions of unpublished GWAS data since 2020.

Published Catalog data is available through our https://www.ebi.ac.uk/gwas/search?query=[search interface], https://www.ebi.ac.uk/gwas/downloads[downloads], https://www.ebi.ac.uk/gwas/docs/api[API] and iconic https://www.ebi.ac.uk/gwas/diagram[GWAS diagram] of SNP-trait associations, mapped onto the human genome by chromosomal location and displayed on the human karyotype. Summary statistics files are available to download from our ftp://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/[FTP site] via links from the study in the https://www.ebi.ac.uk/gwas/search?query=[search interface] or our designated https://www.ebi.ac.uk/gwas/downloads/summary-statistics[summary statistics page]. Harmonised summary statistics are also available from our summary statistics database via https://www.ebi.ac.uk/gwas/summary-statistics/docs/[API]. Unpublished GWAS data is available through a separate https://www.ebi.ac.uk/gwas/downloads[download].

Since 2010, delivery and development of the Catalog has been a collaborative project between the EMBL-EBI and NHGRI. The Catalog website and infrastructure is now hosted by EMBL-EBI. A team of curators, all of whom are experienced geneticists or molecular biologists, curate the data, and provide user support. The curation team is supported by dedicated software developers, and is led by Helen Parkinson at EMBL-EBI and Mike Inouye at the University of Cambridge, and supported by NHGRI staff. We work closely with the https://www.ebi.ac.uk/about/people/helen-parkinson[SPOT] and http://www.ensembl.org/info/about/index.html[Ensembl] teams at the EBI, and receive input from an independent Scientific Advisory Board.

For more details about the Catalog curation process and data extraction procedures, please refer to the link:methods[Methods page].

== Contact 

=== Contacting the GWAS Catalog team

To contact the GWAS Catalog team, with queries, comments or feedback please email gwas-info@ebi.ac.uk. Please see our link:faq[FAQ page] for answers to some common queries about the GWAS Catalog.

To submit data to the GWAS Catalog please visit our https://www.ebi.ac.uk/gwas/deposition[submission page].

For more help using the GWAS Catalog, please have a look at our link:related-resources[training resources].


=== Keeping up-to-date

We announce new releases and other developments on our announcement list, \gwas-announce@ebi.ac.uk. link:++mailto:gwas-announce-join@ebi.ac.uk?subject=subscribe&body=Please subscribe me to receive GWAS Catalog updates++[Sign up for updates].

To interact with other GWAS Catalog users, join our user community mailing list, \gwas-users@ebi.ac.uk. link:++mailto:gwas-users-join@ebi.ac.uk?subject=subscribe&body=Please subscribe me to the GWAS users list++[Send us a quick email] to subscribe to this list.

You can also follow us on https://www.linkedin.com/company/gwas-catalog/[LinkedIn], BlueSky https://bsky.app/profile/gwascatalog.bsky.social[@gwascatalog.bsky.social] and X https://twitter.com/GWASCatalog[@GWASCatalog].


== Funding acknowledgements

The NHGRI-EBI GWAS Catalog is funded by NHGRI Grant Number 1U24HG012542-01, and delivered by collaboration between the NHGRI, EMBL-EBI, University of Cambridge and NCBI. We recieve additional funding from Open Targets. 


== Citing the GWAS Catalog


=== How to cite the NHGRI-EBI GWAS Catalog, data therein or diagram:

_Cerezo M, Sollis E, Ji Y, Lewis E, Abid A, Bircan KO, Hall P, Hayhurst J, John S, Mosaku A, Ramachandran S, Foreman A, Ibrahim A, McLaughlin J, Pendlington Z, Stefancsik R, Lambert SA, McMahon A, Morales J, Keane T, Inouye M, Parkinson H, Harris LW._ +
https://doi.org/10.1093/nar/gkae1070[doi.org/10.1093/nar/gkae1070] +
Nucleic Acids Research, Volume 53, Issue D1, 6 January 2025, Pages D998–D1005 +

Users of summary statistics are requested to cite the data as follows: accession ID of the GWAS Catalog study e.g. “GCST007240”, the GWAS Catalog, and the date the summary statistics were downloaded. If the summary statistics originated from a published GWAS, please also cite the original publication. For example, “Summary statistics were downloaded from the NHGRI-EBI GWAS Catalog (Cerezo et al., 2025) on 01/02/2025 for study GCST007240 (Riveros-McKay et al., 2019).

For a full list of publications by the GWAS Catalog team, and other useful background reading, please visit our link:related-resources[Resources page]


== Licensing information

The NHGRI-EBI GWAS Catalog and all its contents are available under the general http://www.ebi.ac.uk/about/terms-of-use[Terms of Use for EMBL-EBI Services]. Summary statistics are made available under https://creativecommons.org/publicdomain/zero/1.0/[CC0] unless https://www.ebi.ac.uk/gwas/docs/faq#faqH7CC0License[otherwise stated]. We advise consumers of data hosted by the GWAS Catalog to note the license terms of individual datasets, if applicable to their specific use case. 

GWAS Catalog logos, and data visualisations such as the GWAS Diagram and LD plots, are available for re-use under https://creativecommons.org/licenses/by/4.0/[CC-BY-4.0]. 

All code for the GWAS Catalog infrastructure can be found in https://github.com/EBISPOT/goci[our Github repository]. The code is freely available under http://www.apache.org/licenses/LICENSE-2.0[the Apache version 2.0 license]

The GWAS Catalog adheres to the FAIR indicators of Findability, Accessibility, Interoperability, and Reusability, described in more detail in https://www.cell.com/cell-genomics/fulltext/S2666-979X(21)00004-5[MacArthur et al, Workshop proceedings: GWAS Catalog standards and sharing, Cell Genomics (2021)] and registered on https://fairsharing.org/FAIRsharing.blUMRx[fairsharing.org].

== Privacy policy

More details about our privacy policy can be found by following the links below.

https://drive.google.com/file/d/1tqBxiyo6llXgGtED7jiup-PNh50ZvQv3/view?usp=sharing[GWAS Catalog Submissions] applies to the data submitted to the GWAS Catalog (eg. Summary statistics files).

https://drive.google.com/file/d/1TOCBrJnhrRAJ62Rmx3KtqxNbviUm1g9w/view?usp=sharing[GWAS Catalog Mail Services] applies to our public e-mail lists such as gwas-info@ebi.ac.uk, gwas-announce@ebi.ac.uk and gwas-users@ebi.ac.uk.

https://drive.google.com/file/d/18banglYT7UwOJaoIKl9qimobwPysX-MJ/view?usp=sharing[GWAS Catalog Curation Tracking] applies to any personal data acquired from publications, or from communication with authors (including author names, e-mails and addresses), that we retain in order to track GWAS publications through our curation processes as part of making those publications available as part of the GWAS Catalog.

== Credits &amp; acknowledgements

* Font generated by http://www.flaticon.com[flaticon.com] under http://creativecommons.org/licenses/by/3.0/[CC BY license].
The authors are http://picol.org[Picol].

* http://www.jetbrains.com/idea/index.html[IntelliJ IDEA]: Full-featured commercial IDE with a complete set of tools and integrations with the most important modern technologies and frameworks, such as Spring and Hibernate — a must-have for effective Web and Java EE development.


