
= Description of column headings for downloadable files

== Description of column headings for downloadable files


Association data in the GWAS Catalog is mapped to the genome using the latest version of Ensembl. For details on genome assembly and dbSNP build, refer to the GWAS Catalog https://www.ebi.ac.uk/gwas[homepage].


<<File headers for catalog v1.0>>

<<File headers for catalog v1.0.1>>

<<File headers for catalog v1.0.2>>

<<File headers for catalog v1.0.2.1>>

<<File headers for ancestry download>>

<<File headers for unpublished studies v1.0.3>>

<<File headers for unpublished studies v1.0.3.1>>

<<File headers for unpublished ancestries>>

*Available in associations download files

+Available in studies download files

'''

=== File headers for catalog v1.0

DATE ADDED TO CATALOG^*^ ^+^: Date a study is published in the catalog

PUBMEDID^*^ ^+^: PubMed identification number

FIRST AUTHOR^*^ ^+^: Last name and initials of first author

DATE^*^ ^+^: Publication date (online (epub) date if available)

JOURNAL^*^ ^+^: Abbreviated journal name

LINK^*^ ^+^: PubMed URL

STUDY^*^ ^+^: Title of paper

DISEASE/TRAIT^*^ ^+^: Disease or trait examined in study

INITIAL SAMPLE DESCRIPTION^*^ ^+^: Sample size and ancestry description for stage 1 of GWAS (summing across multiple Stage 1 populations, if applicable)

REPLICATION SAMPLE DESCRIPTION^*^ ^+^: Sample size and ancestry description for subsequent replication(s) (summing across multiple populations, if applicable)

REGION^*^: Cytogenetic region associated with rs number

CHR_ID^*^: Chromosome number associated with rs number

CHR_POS^*^: Chromosomal position associated with rs number

REPORTED GENE(S)^*^: Gene(s) reported by author

MAPPED GENE(S)^*^: Gene(s) mapped to the strongest SNP. If the SNP is located within a gene, that gene is listed. If the SNP is located within multiple genes, these genes are listed separated by commas. If the SNP is intergenic, the upstream and downstream genes are listed, separated by a hyphen.

UPSTREAM_GENE_ID^*^: Entrez Gene ID for nearest upstream gene to rs number, if not within gene

DOWNSTREAM_GENE_ID^*^: Entrez Gene ID for nearest downstream gene to rs number, if not within gene

SNP_GENE_IDS^*^: Entrez Gene ID, if rs number within gene; multiple genes denotes overlapping transcripts

UPSTREAM_GENE_DISTANCE^*^: distance in kb for nearest upstream gene to rs number, if not within gene

DOWNSTREAM_GENE_DISTANCE^*^: distance in kb for nearest downstream gene to rs number, if not within gene

STRONGEST SNP-RISK ALLELE^*^: SNP(s) most strongly associated with trait + risk allele (? for unknown risk allele). May also refer to a haplotype.

SNPS^*^: Strongest SNP; if a haplotype it may include more than one rs number (multiple SNPs comprising the haplotype)

MERGED^*^: denotes whether the SNP has been merged into a subsequent rs record (0 = no; 1 = yes;)

SNP_ID_CURRENT^*^: current rs number (will differ from strongest SNP when merged = 1)

CONTEXT^*^: provides information on a variant's predicted most severe functional effect from Ensembl

INTERGENIC^*^: denotes whether SNP is in intergenic region (0 = no; 1 = yes)

RISK ALLELE FREQUENCY^*^: Reported risk/effect allele frequency associated with strongest SNP in controls (if not available among all controls, among the control group with the largest sample size). If the associated locus is a haplotype the haplotype frequency will be extracted.

P-VALUE^*^: Reported p-value for strongest SNP risk allele (linked to dbGaP Association Browser). Note that p-values are rounded to 1 significant digit (for example, a published p-value of 4.8 x 10-7 is rounded to 5 x 10-7).

PVALUE_MLOG^*^: -log(p-value)

P-VALUE (TEXT)^*^: Information describing context of p-value (e.g. females, smokers).

OR or BETA^*^: Reported odds ratio or beta-coefficient associated with strongest SNP risk allele. Note that prior to 2021, any OR <1 was inverted, along with the reported allele, so that all ORs included in the Catalog were >1. This is no longer done, meaning that associations added after 2021 may have OR <1. Appropriate unit and increase/decrease are included for beta coefficients.

95% CI (TEXT)^*^: Reported 95% confidence interval associated with strongest SNP risk allele, along with unit in the case of beta-coefficients. If 95% CIs are not published, we estimate these using the standard error, where available.

PLATFORM (SNPS PASSING QC)^*^: Genotyping platform manufacturer used in Stage 1; also includes notation of pooled DNA study design or imputation of SNPs, where applicable

CNV^*^: Study of copy number variation (yes/no)

ASSOCIATION COUNT^+^: Number of associations identified for this study

_Last updated: 12 January 2016_

'''

=== File headers for catalog v1.0.1

As for version 1.0 plus

MAPPED_TRAIT^*^ ^+^: Mapped Experimental Factor Ontology trait for this study

MAPPED_TRAIT_URI^*^ ^+^: URI of the EFO trait

STUDY ACCESSION^*^ ^+^: Accession ID allocated to a GWAS Catalog study

Dismissed from 1 May 2018

_Last updated: 24 April 2018_

'''

=== File headers for catalog v1.0.2

As for version 1.0.1 plus

GENOTYPING_TECHNOLOGY^*^ ^+^: Genotyping technology/ies used in this study, with additional array information (ex. Immunochip or Exome array) in brackets.

_Added: 24 April 2018_

'''

=== File headers for catalog v1.0.2.1
As for version 1.0.2 plus

COHORT+: Discovery stage cohorts used in this study. The full list of cohort abbreviations and definitions can be found https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata/pgs_all_metadata_cohorts.csv[here]. 

FULL SUMMARY STATISTICS+: Availability of full genome-wide summary statistics files for download 

SUMMARY STATS LOCATION+: The location of the summary statistics file

_Added: October 2023_

'''

=== File headers for ancestry download

STUDY ACCESSION: Accession ID allocated to a GWAS Catalog study

PUBMEDID: PubMed identification number

FIRST AUTHOR: Last name and initials of first author

DATE: Publication date (online (epub) date if available)

INITIAL SAMPLE DESCRIPTION: Sample size and ancestry description for GWAS stage (summing across multiple populations, if applicable)

REPLICATION SAMPLE DESCRIPTION: Sample size and ancestry description for subsequent replication(s) (summing across multiple populations, if applicable)

STAGE: Stage of the GWAS to which the sample description applies, either initial or replication

NUMBER OF INDIVDUALS: Number of individuals in this sample

BROAD ANCESTRAL CATEGORY: Broad ancestral category to which the individuals in the sample belong

COUNTRY OF ORIGIN: Country of origin of the individuals in the sample

COUNTRY OF RECRUITMENT: Country of recruitment of the individuals in the sample

ADDITONAL ANCESTRY DESCRIPTION: Any additional ancestry descriptors relevant to the sample description

_Added: 16 September 2016_

'''

=== File headers for unpublished studies v1.0.3

DATE ADDED TO CATALOG	Not used

PUBMED ID	Not used

FIRST AUTHOR	Last name and initials of first author

DATE	Not used

JOURNAL	Not used

LINK	URL of the manuscript the GWAS is included in

STUDY	Title of manuscript the GWAS is included in

DISEASE/TRAIT	Disease or trait examined in the GWAS

INITIAL SAMPLE SIZE	Not used

REPLICATION SAMPLE SIZE	Not used

PLATFORM [SNPS PASSING QC]	Genotyping platform manufacturer and number of SNPs tested in the analysis; also includes imputation of SNPs, where applicable

ASSOCIATION COUNT	Not used

MAPPED_TRAIT	Not used

MAPPED_TRAIT_URI	Not used

STUDY ACCESSION	GWAS Catalog Accession ID

GENOTYPING TECHNOLOGY	Genotyping technology/ies used in this study, with additional array information (eg. Immunochip or Exome array) in brackets.

BACKGROUND TRAIT	Any background trait(s) shared by all individuals in the GWAS

MAPPED BACKGROUND TRAIT	Not used

MAPPED BACKGROUND TRAIT URI	Not used

SUMMARY STATS LOCATION	The location of the summary statistics file

SUBMISSION DATE	The date the GWAS was submitted to the Catalog

STATISTICAL MODEL	Details of the statistical model used to determine association significance

_Added: 15 May 2020_

'''

=== File headers for unpublished studies v1.0.3.1

As for version 1.0.3 plus

COHORT+: Discovery stage cohorts used in this study. The full list of cohort abbreviations and definitions can be found https://ftp.ebi.ac.uk/pub/databases/spot/pgs/metadata/pgs_all_metadata_cohorts.csv[here]. 

FULL SUMMARY STATISTICS+: Availability of full genome-wide summary statistics files for download

_Added October 2023_ 

'''

=== File headers for unpublished ancestries

STUDY ACCESSION	GWAS Catalog Accession ID

PUBMED ID	not used

FIRST AUTHOR	Last name and initials of first author

DATE	not used

INITIAL SAMPLE DESCRIPTION	not used

REPLICATION SAMPLE DESCRIPTION	not used

STAGE	not used/filled with initial for all

NUMBER OF INDIVIDUALS	Number of individuals in this sample

BROAD ANCESTRAL CATEGORY	Broad ancestral category to which the individuals in the sample belong

COUNTRY OF ORIGIN	not used

COUNTRY OF RECRUITMENT	Country of recruitment of the individuals in the sample

ADDITIONAL ANCESTRY DESCRIPTION	Any additional ancestry descriptors relevant to the sample description

ANCESTRY DESCRIPTOR	The most detailed ancestry descriptor(s) for the sample.

FOUNDER/GENETICALLY ISOLATED POPULATION	Description of a founder or genetically isolated population

NUMBER OF CASES	The number of cases in this broad ancestry group

NUMBER OF CONTROLS	The number of controls in this broad ancestry group

SAMPLE DESCRIPTION	Additional sample information required for the interpretation of result

COHORT(S)	List of cohort(s) represented in the discovery sample (moved to studies download in v1.0.3.1)

COHORT-SPECIFIC REFERENCE	List of cohort specific identifier(s) issued to this research study (removed in v1.0.3.1)

_Added 15 May 2020, updated October 2023_

'''

