= Training & Related resources

1. <<section-1, GWAS Catalog training materials>>
2. <<section-2, Publications>>
3. <<section-3, Useful resources>>
4. <<section-4, Community workshops>>

=== [[section-1]]GWAS Catalog training materials

==== EBI Train online

New to the Catalog? Start here with link:https://www.ebi.ac.uk/training/online/courses/gwas-catalogue-exploring-snp-trait-associations[GWAS Catalog - Exploring SNP-trait associations], an easily accessible interactive tutorial on the full range of features the GWAS Catalog has to offer.

==== Training videos

Recorded webinars are available to guide you through different aspects of the Catalog:

* https://www.ebi.ac.uk/training/events/exploring-genotype-phenotype-data-using-gwas-catalog/[Exploring genotype-phenotype data using the GWAS Catalog] (2022) for an introduction to the GWAS Catalog and a demonstration of our web-based search interface.

* https://www.ebi.ac.uk/training/events/data-submission-genome-wide-association-studies-gwas-catalog/[Data submission for genome-wide association studies at the GWAS Catalog] (2023) for a detailed walkthrough of our summary statistics submission system.

Earlier webinars are also available:

* https://www.ebi.ac.uk/training/events/investigating-genotype-phenotype-data-using-gwas-catalog/[Investigating genotype/phenotype data using the GWAS Catalog] (2020).

==== In-person training materials

Training materials from in-person training workshops held by the GWAS Catalog team are available on link:https://github.com/EBISPOT/eshg-2024-workshop[Github] and link:ftp://ftp.ebi.ac.uk/pub/databases/gwas/training_materials[our FTP server]. If you would like to make use of these slides, please cite the GWAS Catalog as detailed on the link:about[About] page.


=== [[section-2]]Publications

==== GWAS Catalog publications

**Sollis et al, 2023 https://doi.org/10.1093/nar/gkac1010[The NHGRI-EBI GWAS Catalog: knowledgebase and deposition resource]. Nucleic Acids Research 51(D1):D977-D985**

Stefancik et al, 2023 https://link.springer.com/article/10.1007/s00335-023-09992-1[The Ontology of Biological Attributes (OBA) - Computational Traits for the Life Sciences]. Mammalian Genome 34, 364–378.

MacArthur et al, 2021 https://www.cell.com/cell-genomics/fulltext/S2666-979X(21)00004-5[Workshop proceedings: GWAS Catalog standards and sharing]. Cell Genomics Vol. 1, Issue 1, 100004.

McMahon et al, 2021 https://www.cell.com/cell-genomics/fulltext/S2666-979X(21)00005-7[Sequencing-based genome-wide association studies reporting standards]. Cell Genomics Vol. 1, issue 1, 100005.

Lambert et al, 2021 https://www.nature.com/articles/s41588-021-00783-5/[The Polygenic Score Catalog as an open database for reproducibility and systematic evaluation]. Nature Genetics volume 53, pages420–425.
  
Buniello, MacArthur et al, 2019 link:https://academic.oup.com/nar/article/47/D1/D1005/5184712[The NHGRI-EBI GWAS Catalog of published genome-wide association studies, targeted arrays and summary statistics 2019]. Nucleic Acids Research, 2019, Vol. 47 (Database issue): D1005-D1012.

Lee K et al, 2018 link:https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1006390[Scaling up data curation using deep learning: An application to literature triage in genomic variation resources].  PLoS Computational Biology 14(8):e1006390.

Morales et al, 2017 link:http://rdcu.be/G6Fv[A standardized framework for representation of ancestry data in genomics studies, with application to the NHGRI-EBI GWAS Catalog]. Genome Biology, 19:21.

MacArthur et al, 2017 link:https://academic.oup.com/nar/article/45/D1/D896/2605751[The new NHGRI-EBI Catalog of published genome-wide association studies (GWAS Catalog)]. Nucleic Acids Research, Vol. 45 (Database issue): D896-D901.

Welter D et al, 2014 link:https://academic.oup.com/nar/article/42/D1/D1001/1062755[The NHGRI GWAS Catalog, a curated resource of SNP-trait associations]. Nucleic Acids Research, Vol. 42 (Database issue): D1001-D1006.

Hindorff LA, et al 2009 link:http://www.genome.gov/Pages/About/OD/NewsAndFeatures/PNASGWASOnlineCatalog.pdf[Potential etiologic and functional implications of genome-wide association loci for human diseases and traits]. Proc. Natl. Acad. Sci. USA, 2009, 106, 9362–9367.

==== Other background reading

Uffelmann et al, 2021 https://www.nature.com/articles/s43586-021-00056-9[Genome-wide association studies]. Nat Rev Methods Primers 1, 59.

link:https://www.yourgenome.org/stories/genome-wide-association-studies[A beginners guide to GWAS] (2016)

link:http://ontogenesis.knowledgeblog.org/66[What is an Ontology?] (2010)

Visscher et al, 2017 link:http://www.sciencedirect.com/science/article/pii/S0002929717302409?via%3Dihub[10 Years of GWAS Discovery: Biology, Function and Translation] AJHG Volume 101, Issue 1, 6 July 2017, Pages 5-22

Manolio, 2010 link:http://www.nejm.org/doi/full/10.1056/NEJMra0905980[Genomewide Association Studies and assessment of the risk of disease] NEJM 363:166-176

==== Blog posts

https://ebispot.github.io/gwas-blog/background-trait-update[Trait annotation in the GWAS Catalog] (2021): a discussion of how traits are annotated in the GWAS Catalog, focussing on improved background trait annotation.

http://blog.opentargets.org/2021/03/11/open-sharing-of-cancer-summary-statistics/[Sharing is caring: why we need more freely available cancer GWAS summary statistics] (2021): a blog post calling for more freely available cancer GWAS summary statistics.

http://www.ensembl.info/2018/06/28/ten-years-of-the-gwas-catalog-past-present-and-future/[Ten years of the GWAS Catalog – Past, present and future](2018): 10th Anniversary blog post highlighting the links between the GWAS Catalog and Ensembl.

http://blog.opentargets.org/2018/06/13/everything-you-need-to-know-about-the-new-gwas-catalog/[Everything you need to know about the new GWAS Catalog] (2018): a blog post describing the collaboration between the GWAS Catalog and Open Targets.

Or see the https://ebispot.github.io/gwas-blog/[full index].

=== [[section-3]]Useful resources

==== Tools & data services

// tag::b-col-indv[]
[cols="1,2"]
|===

|https://github.com/ramiromagno/gwasrapidd[gwasrapidd]
|an R package to query, download and wrangle GWAS Catalog data

|https://pypi.org/project/pandasgwas/[pandasgwas]
|A Python package for retrieval of GWAS Catalog data

|https://www.ensembl.org/info/docs/tools/vep/index.html[Variant Effect Predictor (VEP)]
|Determines the effect of variants (SNPs, insertions, deletions, CNVs or structural variants) on genes, transcripts, and protein sequence, as well as regulatory regions

|http://www.phenoscanner.medschl.cam.ac.uk/phenoscanner[Phenoscanner]
|Tool to facilitate “phenome scans”, the cross-referencing of genetic variants with a broad range of phenotypes, using a range of data sources including GWAS, eQTL and metabolite data

|https://phgkb.cdc.gov/PHGKB/hNHome.action[HuGE Navigator]
|Knowledge base in human genome epidemiology, including information on population prevalence of genetic variants, gene-disease associations, gene-gene and gene- environment interactions

|http://www.ncbi.nlm.nih.gov/gap/PheGenIUCSC[Phenotype-Genotype Integrator]
|Merges GWAS Catalog Data with other NCBI databases to facilitate prioritisation of GWAS hits for follow-up

|http://www.pgscatalog.org/[PGS Catalog]
|An open database of polygenic scores and the relevant metadata required for accurate application and evaluation

|http://targetvalidation.org[Open Targets Platform]
|Platform for the identification of drug targets using multiple data types including GWAS

|https://unmtid-shinyapps.net/shiny/tiga/[Target Illumination GWAS Analytics (TIGA)]
|Resource for drug target illumination by scoring and ranking protein-coding genes associated with traits from GWAS

|link:http://locuszoom.org/[LocusZoom]
|A suite of software for creating regional association plots from GWAS data

|https://bio.tools/?page=1&q=GWAS%20study&sort=score&ord=desc[Elixir Tools and Data Services Registry]
|A registry of GWAS-related resources and tools

|https://omictools.com/search?q=GWAS[OMICStools]
|Searchable database of GWAS and other bioinformatics tools

|https://pharos.nih.gov/idg/index[Pharos]
|Pharos is the user interface to the Knowledge Management Center (KMC) for the Illuminating the Druggable Genome (IDG) program

|https://gwasdiversitymonitor.com[GWAS Diversity Monitor]
|Interactive dashboard monitoring the diversity of participants across all published GWAS|

|===
// end::b-col-indv[]

==== Resources used within the GWAS Catalog

For more information on how we use these resources, see our link:https://www.ebi.ac.uk/gwas/docs/methods[Methods page].

// tag::b-col-indv[]
[cols="1,2"]
|===

|link:https://www.ebi.ac.uk/efo/[Experimental Factor Ontology] (EFO)
|The ontology used for trait mapping in the GWAS Catalog

|link:http://www.ensembl.org/[Ensembl Genome Browser]
|Genome browser and data for vertebrate genomes, used in our variant mapping pipeline. Ensembl annotates genes, computes multiple alignments, predicts regulatory function and collects disease data

|link:https://www.ncbi.nlm.nih.gov/pubmed[PubMed]
|Literature search and citation annotation

|link:http://europepmc.org/[EuropePMC]
|Literature search and citation annotation

|link:https://www.cia.gov/library/publications/the-world-factbook/geos/lg.html[CIA Factbook]
|A rich source of population demographic data, used by curators

|link:https://www.ncbi.nlm.nih.gov/[NCBI]
|Data used for variant mapping in older versions of the Catalog

|===
// end::b-col-indv[]


==== Genome and phenome resources

// tag::b-col-indv[]
[cols="1,2"]
|===

|link:http://www.internationalgenome.org/[1000 Genomes]
|The largest public reference catalogue of worldwide human variation and genotype data

|http://www.hapmap.org/[International HapMap Project]
|The first public catalogue of human variation data, now superseded by 1000Genomes

|http://genome.ucsc.edu/cgi-bin/hgGateway[UCSC Genome Browser]
|Genome browser to interactively visualize genomic data

|link:http://www.phenxtoolkit.org/[PhenX toolkit]
|Consensus measures of Phenotypes and eXposures

|http://www.ncbi.nlm.nih.gov/projects/SNP/[dbSNP]
|Database of human single nucleotide polymorphisms

|https://www.ncbi.nlm.nih.gov/[National Centre for Biotechnology Information]
|Provides wide range of resources including Entrez gene, GenBank, PubMed and ClinVar

|===
// end::b-col-indv[]


==== Sources of summary statistics

Summary statistics are available via the https://www.ebi.ac.uk/gwas/downloads/summary-statistics[GWAS Catalog] where possible. Find additional sources of summary statistics https://www.ebi.ac.uk/gwas/downloads/summary-statistics[here].

=== [[section-4]]Community workshops

Videos from our community workshop on **standards and infrastructure for predicted effector gene lists**, held in September 2024 as a collaboration between the GWAS Catalog, https://kp4cd.org/[HuGEAMP Knowledge Portal Network] and https://www.opentargets.org/[Open Targets], can be found https://www.youtube.com/@hugeampknowledgeportals1943[here]. 

Materials from our community workshop on **GWAS summary statistics standards and sharing**, held in June 2020 to address the standards, infrastructure and incentives required to promote and enable broader sharing of data, are available https://www.ebi.ac.uk/gwas/docs/sharing-standards-workshop[here] and are published as MacArthur et al, 2021 https://www.cell.com/cell-genomics/fulltext/S2666-979X(21)00004-5[Workshop proceedings: GWAS Catalog standards and sharing], Cell Genomics 1(1). 


