= Preparing summary statistics files


Summary statistics are defined as the aggregate p-values and association data for every variant analysed in a genome-wide association study (GWAS).


The GWAS Catalog maintains a standard format for summary statistics, to ensure that datasets from a wide range of sources are as interoperable as possible. This standard format has been developed in consultation with the GWAS research community, and may be updated in future as the needs of the community change.


The current version of the standard format is *GWAS-SSF* (see https://github.com/EBISPOT/gwas-summary-statistics-standard[detailed specifications and citation information^]).


All submitted summary statistics are required to conform to the standard format. If you have difficulty meeting the requirements, please contact gwas-subs@ebi.ac.uk. This page outlines how to prepare and validate summary statistics files, including:


* <<validator, GWAS SumStats Tools (formatter and validator)>>
* <<format, Summary statistics file format and content>>
* <<readme, README content>>
* <<checksums, Summary statistics checksums>>


== [[validator]]GWAS SumStats Tools (formatter and validator)

GWAS SumStats Tools is a versatile toolkit created to simplify the processing, validation, and formatting of GWAS summary statistics files intended for submission to the GWAS Catalog: 

* The formatter tool provides a number of useful functions to edit a summary statistics file, such as splitting, renaming and reordering columns; and finding, replacing and extracting specific patterns across all values in a column.

* The validator tool can be run prior to submission, to ensure that you have prepared your files in the standard format.

Both tools can be run locally via the command line or through the https://ebispot.github.io/gwas-sumstats-tools-ssf-morph/[SSF-morph^] web interface. For set-up instructions and further details, please see https://ebispot.github.io/gwas-sumstats-tools-documentation/#/README[the GWAS SumStats Tools documentation^].

== [[format]]Summary statistics file format and content


* Summary statistics should be provided in a separate file per GWAS.
* The file must be in .tsv format.
* The first line contains the table headers, followed by one row for each variant analysed.
* Columns must follow the standard headers and data formats listed in Table 1.
* All mandatory fields must be present, in the defined order (0-7) (Table 1).
* Table 1 shows some non-mandatory (encouraged) fields that may be present (*). You are encouraged to include these additional fields in the standard format, where possible.
* You may include any number of additional columns beyond the set of mandatory fields. 
* If the output software or analysis method precludes submission of beta/OR/HR and/or standard error, please get in touch to discuss alternative submission arrangements
* We recommend that analyses are performed with effect allele and other allele fixed to ALT and REF where possible, which provides optimal identification of indels. However, other allele orders can be submitted. The optional column ref_allele should be used to indicate which allele is REF (see Table 1).
* We request you submit files that correspond to the final QC’d version of your dataset, including only data which is usable in downstream analyses (e.g. applying filters for MAF, imputation quality, variant present in n cohorts, as you would do in your own research). We do not apply any of these filters post-submission.



*Table 1. Required content and format*.
Mandatory and encouraged fields are shown, with standard headers.


[cols="<1,<2,<3,<2,<3,<1", options="header", grid="all", width=100%]
|===
|Field order
|Field name
|Description
|Type
|Accepted values
|Example

|0
|chromosome
|Chromosome where the variant is located
|Mandatory
|`1-22`, `23`=`X`, `24`=`Y` or `25`=`MT`
|`13`

|1
|base_pair_location
|The first position of the variant in the reference, counting on the bases, from 1
|Mandatory
|Integer > 0
|`40983974`

|2
|effect_allele
|The allele associated with the effect
|Mandatory
|Sequence of `A`, `C`, `T` and `G`.
|`G`

|3
|other_allele
|The non-effect allele
|Mandatory
|Sequence of `A`, `C`, `T` and `G`
|`A`

|4
|(see below)
|Field 4 must contain the primary effect size measurement (i.e. beta, odds ratio _or_ hazard ratio). If you have calculated more than one effect size statistic (e.g. beta as well as odds ratio), the other can be entered as an additional column at the end of the file.
|Mandatory to include beta _or_ odds_ratio _or_ hazard_ratio
|-
|-

|4
|beta
|Effect measured as beta
|(see above)
|Number
|`-0.064`

|4
|odds_ratio
|Effect measured as odds ratio
|(see above)
|Number ≥ 0
|`1.50`

|4
|hazard_ratio
|Effect measured as hazard ratio
|(see above)
|Number ≥ 0
|`0.70`


|5
|standard_error
|Standard error of the effect
|Mandatory
|Number
|`0.03`


|6
|effect_allele_frequency
|Frequency of the effect allele in the control population
|Mandatory
|Number between 0 and 1. If required, effect allele frequency may be rounded to a threshold value defined in the metadata
|`0.22`

|7
|(see below)
|Column 7 must contain the significance value (i.e. p_value _or_ neg_log_10_p_value). If you provide the p_value as a negative log10 value, please make sure to change the column header to match.
|Mandatory to include p_value _or_ neg_log_10_p_value
|-
|-

|7
|p_value
|P-value of the association statistic
|(see above)
|Number between 0 and 1 (standard p-value). If any p-value is equal to 0, the precision of the p-value must be given in the accompanying metadata
|`1E-08`

|7
|neg_log_10_p_value
|Negative log10 p-value of the association statistic
|(see above)
|Number ≥ 0 (-log10 p-value).
|`8.00`

|*
|variant_id
|An internal variant identifier in the form of 
`<chromosome>_<base_pair_location>_
<reference_allele>_<alternate_allele>`
|Encouraged
|See description, concatenated with underscores (`_`).
`LONG_STRING` may be used where allele string is too long to be represented.
|`13_40983974_A_G`


|*
|rs_id
|The rsID of the variant
|Encouraged
|Valid rsID
|`rs7329174`


|*
|info
|Imputation information metric
|Encouraged
|Number between 0 and 1
|`0.98`


|*
|ci_lower
|Lower confidence interval for the odds ratio
|Encouraged
|Number
|`1.36`


|*
|ci_upper
|Upper confidence interval for the odds ratio
|Encouraged
|Number
|`1.64`


|*
|ref_allele
|State which of the alleles is the reference allele
|Encouraged
|`EA` for effect allele, `OA` for other allele or `#NA` if unknown
|`EA`


|*
|n
|Sample size per variant
|Encouraged
|Integer
|`15000`
|===


== [[readme]]README content


Each summary statistics file should be accompanied by readme text in a standard format. This should be copied into the submission form that you will be able to access during the submission process. Please do not upload a separate readme file.  The readme text should contain the following information in a numbered list:


1. Full citation for the publication linked to the submitted summary statistics (in any standard citation style)
e.g: Klimentidis YC, Raichlen DA, Bea J, Garcia DO, Wineinger NE, Mandarino LJ, Alexander GE, Chen Z, Going SB (2018)
Genome-wide association study of habitual physical activity in over 377,000 UK Biobank participants identifies multiple variants including CADM2 and APOE. International Journal of Obesity. 42: 1161-1176. (If your submission is not for a published article, you may use whatever citation is most appropriate).


2. Clear description of the content presented in additional columns that are not present in our standard format
e.g.: HetPVal: P-value for heterogeneity statistic


3. Any other information or comment that can be useful to interpret and store the data under submission (e.g. broader description of the cohort used in the study, QC measures applied)


== [[checksums]]Summary statistics checksums


In order to ensure the integrity of the uploaded summary statistics files before and after the upload, you must provide an md5 checksum (a kind of “fingerprint”) for each uploaded file. If there is any problem during the upload that leads to corruption of the data, this checksum will change. So by validating against the provided checksum, we can ensure the upload was successful and the data is not corrupted. Although there are various ways to calculate checksums, we are only supporting the most widely used md5 algorithm.


=== Calculating on Mac computers


The md5 command line application is installed on Mac computers by default. Type the following into the command line, replacing “summary_stats.gzip” with the name of your own file:


  $ md5 summary_stats.gzip


The output should look something like this:
 
  MD5 (summary_stats.gzip) = 49ea8cf53801c7f1e2f11336fb8a29c8


The md5 checksum is the 32-digit hexadecimal number in the output, after the filename. Copy this number into the corresponding column of the study sheet.


=== Calculating on Linux/Unix computers


The md5sum command line application is installed on all Linux/Unix based systems by default. Type the following into the command line, replacing “summary_stats.gzip” with the name of your own file:


  $ md5sum summary_stats.gzip


The output should look something like this:


  4ab6d4f4db143eed49c248d3ce23cb57  summary_stats.gzip


The md5 checksum is the 32-digit hexadecimal number in the first column of the output, before the file name. Copy this number into the corresponding column of the study sheet.


=== Calculating on Windows computers


In Windows, you can use the certutil command line tool to calculate md5 checksums. Type the following into the command line, replacing “summary_stats.gzip” with the name of your own file:


  > certutil -hashfile summary_stats.gzip MD5


The output should look something like this:


  > MD5 hash of summary_stats.gzip: d19bbbed9d713f97f487b9ed9ec3f62f


The md5 checksum is the 32-digit hexadecimal number in the output, after the file name. Copy this number into the corresponding column of the study sheet.
