= Preparing summary statistics files

== Summary statistics file format and content

Summary statistics are defined as the aggregate p-values and association data for every variant analysed in a genome-wide association study (GWAS). They should be provided in one file per GWAS with one row for each variant analysed.

Submitted summary statistics files must conform to our standard file format and content (published in https://doi.org/10.1093/nar/gky1120[Buniello and MacArthur, 2019], further details https://www.ebi.ac.uk/gwas/docs/methods/summary-statistics[here].

Summary statistics file requirements:

1. The file must be in .tsv format
2. Columns must follow the standard headings and data formats listed in the table below
3. The file must include at least one of the following combinations of mandatory columns(*): 
  * Option 1: *variant_ID* and *p-value*
  * Option 2: *chromosome*, *base pair location* and *p-value*  - for this option you must also provide the genome assembly in the submission template
4. When one of the manditory columns is not provided, the column should be filled with NA
5. To allow your data to be harmonised after submission (all genomic position data reported against the same and latest genome build and alleles orientated to the forward strand), you should also include *effect_allele* and *other_allele*. See https://www.ebi.ac.uk/gwas/docs/methods/summary-statistics[here] for more details on our harmonisation pipeline.
6. You may also include any number of additional columns

[cols="<2,<3,<1,<3,<1", options="header", grid="all", width=100%]
|===
|Column header
|Description
|Mandatory
|Validation
|Example

|variant_id
|The rsid of the variant
|yes*
|Valid rsID
|`rs7329174`

|p_value
|The p-value of the variant/trait association
|yes*
|Numerical value in range 0 to 1
|`1E-08`

|chromosome
|The chromosome that the variant is located on
|yes*
|Valid chromosome values are 1-25, X, Y, MT (23, 24, 25 are synonymous with ‘X’, ‘Y’, ‘MT’, respectively)
|`13`

|base_pair_location
|The base pair location of the variant
|yes*
|Integer in range 1 to 999999999
|`40983974`

|effect_allele
|The effect allele of the variant
|no
|Valid DNA string
|`G`

|other_allele
|The reference allele
|no
|Valid DNA string
|`A`

|effect_allele_frequency
|Frequency of the effect allele in the control population
|no
|Numerical value in range 0 to 1
|`0.22`

|odds_ratio
|The odds ratio of the variant/trait association
|no
|Numerical value
|`1.50`

|ci_lower
|The lower range of the odds ratio confidence interval
|no
|Numerical value
|`1.36`

|ci_upper
|The upper range of the odds ratio confidence interval
|no
|Numerical value
|`1.64`

|beta
|The beta of the variant/trait association
|no
|Numerical value (+/-)
|`-0.064`

|standard_error
|The standard error of the beta
|no
|Numerical value
|`0.03`
|===

== [[readme]] README file content

Each summary statistics file should be accompanied by a readme file containing the following info in a numbered list:

1. Full citation for the publication linked to the submitted summary statistics (in any standard citation style) 
e.g: Klimentidis YC, Raichlen DA, Bea J, Garcia DO, Wineinger NE, Mandarino LJ, Alexander GE, Chen Z, Going SB (2018) 
Genome-wide association study of habitual physical activity in over 377,000 UK Biobank participants identifies multiple variants including CADM2 and APOE. International Journal of Obesity. 42: 1161-1176.

2. Clear description of the content presented in additional columns that are not present in our standard format
e.g.: INFO: imputation quality/certainty metric from IMPUTE2

3. Any other information or comment that can be useful to interpret and store the data under submission (e.g. broader description of the cohort used in the study, QC measures applied) 

Please see <link to the FTP with example txt file>[here] for an example of a good readme file.

== Summary statistics validator

To validate your files prior to upload, we provide a validator that can be run locally. Using the validator ensures that you have prepared your files in the standard format. For more information on the summary statistics file validator, and how to run it locally, please see: https://pypi.org/project/ss-validate/.

== [[checksums]] Summary statistics checksums

In order to ensure the integrity of the uploaded summary statistics files before and after the upload, you must provide an md5 checksum (a kind of “fingerprint”) for each uploaded file. If there is any problem during the upload that leads to corruption of the data, this checksum will change. So by validating against the provided checksum, we can ensure the upload was successful and the data is not corrupted. Although there are various ways to calculate checksums, we are only supporting the most widely used md5 algorithm.

=== Calculating on Mac computers

+++ <details><summary> +++
+++ </summary><div> +++

The md5 command line application is installed on Mac computers by default. Type the following into the command line, replacing “summary_stats.gzip” with the name of your own file:

  $ md5 summary_stats.gzip

The output should look something like this:
  
  MD5 (summary_stats.gzip) = 49ea8cf53801c7f1e2f11336fb8a29c8

The md5 checksum is the 32-digit hexadecimal number in the output, after the filename. Copy this number into the corresponding column of the study sheet.

+++ </div></details> +++

=== Calculating on Linux/Unix computers

+++ <details><summary> +++
+++ </summary><div> +++

The md5sum command line application is installed on all Linux/Unix based systems by default. Type the following into the command line, replacing “summary_stats.gzip” with the name of your own file:

  $ md5sum summary_stats.gzip

The output should look something like this:

  4ab6d4f4db143eed49c248d3ce23cb57  summary_stats.gzip

The md5 checksum is the 32-digit hexadecimal number in the first column of the output, before the file name. Copy this number into the corresponding column of the study sheet.

+++ </div></details> +++

=== Calculating on Windows computers

+++ <details><summary> +++
+++ </summary><div> +++

In Windows, you can use the certutil command line tool to calculate md5 checksums. Type the following into the command line, replacing “summary_stats.gzip” with the name of your own file:

  > certutil -hashfile summary_stats.gzip MD5

The output should look something like this:

  ...

The md5 checksum is the 32-digit hexadecimal number provided by the application. Copy this number into the corresponding column of the study sheet.

+++ </div></details> +++
