<!DOCTYPE html>
<html lang="en"
      xmlns:th="http://www.thymeleaf.org"
      xmlns="http://www.w3.org/1999/xhtml">

<head>
    <meta charset="utf-8"/>
    <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
    <meta name="viewport" content="width=device-width, initial-scale=1"/>
    <meta name="description"
          content="The NHGRI-EBI GWAS Catalog: a curated collection of all published genome-wide association studies, produced by a collaboration between EMBL-EBI and NHGRI"/>
    <meta name="keywords" content="GWAS Catalog, GWAS, NHGRI, EBI, EMBL-EBI, SPOT"/>
    <meta name="author" content="Tony Burdett, Emma Hastings, Dani Welter, SPOT, EMBL-EBI, NHGRI"/>
    <link rel="icon" th:href="@{/images/favicon.ico}"/>

    <title>GWAS Catalog</title>

    <!-- Bootstrap core CSS -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.1/css/bootstrap.min.css"/>
    <link href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.11.2/themes/smoothness/jquery-ui.css" rel="stylesheet"/>
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/bootstrap-table/1.5.0/bootstrap-table.min.css"/>
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.1/css/bootstrap-theme.min.css"/>

    <!-- Custom styles for this template -->
    <link th:href="@{/css/jumbotron.css}" rel="stylesheet"/>
    <link th:href="@{/css/sticky-footer-navbar.css}" rel="stylesheet"/>
    <link th:href="@{/css/datepicker.css}" rel="stylesheet"/>

    <!-- Additional styling on top of bootstrap -->
    <link rel="stylesheet" th:href="@{/css/goci-ui.css}"/>
    <link rel="stylesheet" th:href="@{/css/ontology-autocomplete.css}"/>
    <link rel="stylesheet" th:href="@{/css/icons/flaticon.css}"/>
    <link rel="stylesheet" th:href="@{/css/goci-color-palette-1.css}"/>


    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
    <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
    <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->

    <script  th:src="@{/js/console-plugin.js}"></script>
</head>

<body>

<div th:include="fragments/header :: navbar"></div>

<div th:include="fragments/header :: jumbotron"></div>

<div class="container-fluid tight-to-top">
    <ol class="breadcrumb background-color-complementary-accent">
        <li><a th:href="@{/home}">Home</a></li>
        <li><a th:href="@{/docs}">Documentation</a></li>
        <li class="active">Summary statistics</li>
    </ol>

    <div>
<div>

</div>
        <!--
        #####################################
        #  Page specific content goes here  #
        #####################################
        -->

        <div style="display: none">
            <ul id="variables">
                <li id="query">User query</li>
            </ul>
        </div>

        <div class="container">
            <!--div class="row">
                <h2 style="color: black">GWAS Summary Statistics</h2>
                <p>
                    There are thousands of genome-wide association studies and each study yields association data for hundreds of thousands of variants across the human genome.
                    Manual curation of each GWAS publication by a dedicated team of scientists ensures that the Catalog contains the most significant findings (p-value &lt; 10-5).
                </p>

                <p>
                    Studies are often accompanied with summary statistics providing the association data for all the variants analysed across the genome in a given study. Although most of these associations will be insignificant or ‘noise’, there will also be some highly significant associations. We aim to provide the tools to access these data and we encourage authors of GWAS publications to submit <a th:href="@{/docs/methods/summary-statistics}">their summary statistics to us</a> so that we can develop this into a powerful resource.
                </p>

                <p>
                    Our first step to making summary statistics data accessible was by hosting the flat files on our FTP site for those studies that provided them. The main challenges for this type of data acquisition are that it is difficult to search and filter, the volume of data is prohibitively large and the data are not harmonised (see below). To address these challenges, we have now developed a dedicated summary statistics database, providing users with searchable, filterable, harmonised data via a <a href="https://www.ebi.ac.uk/gwas/summary-statistics/docs/" target="_blank">REST API.</a>For non-programmatic access, data are available on the <a th:href="@{/downloads/summary-statistics}">FTP site</a>.
                </p>

                <p>
                    To make these data available in a consistent, searchable format, we had to define a summary statistics standard because there is no standard in place. Standardisation resolves the issues of files from different research groups containing different fields and headers to one another, there is no consensus on what should be included, what should be omitted and what data format to use. An additional challenge to users of these data is that the variants are reported against different genome builds, so the location of the same variant can be different between studies. Even if the genome build was the same, the same variant may be reported against the other strand from one study to the next. Furthermore, what is taken to be the effect allele one the study may the ancestral allele in others. To mitigate these inconsistencies all of the summary statistics data is harmonised {link}. For information on <a th:href="@{/docs/methods/summary-statistics}">the harmonisation see</a>.
                </p>

            </div-->
            <div class="row">
                <a th:href="@{/docs/methods/summary-statistics}">
                    <div class="col-md-4 toplevel-view">
                        <h2 class="flaticon-information9 color-primary-bold">Documentation</h2>

                        <p>Summary statistics eligibility, submission information, standards and harmonisation.</p>
                    </div>
                </a>
                <a th:href="@{/summary-statistics/docs/}" target="_blank">
                    <div class="col-md-4 toplevel-view">
                        <h2 class="flaticon-source2 color-primary-bold">API</h2>

                        <p>A guide to how to access summary statistics data programmatically.</p>
                    </div>
                </a>
                <a th:href="@{/downloads/summary-statistics}">
                    <div class="col-md-4 toplevel-view">
                        <h2 class="flaticon-statistics color-primary-bold">Available studies</h2>

                        <p>A list of all studies for which summary statistics are available in the Catalog.</p>
                    </div>
                </a>

            </div>
<!--            <div class="row">-->
<!--                <a th:href="@{${@environment.getProperty('deposition.ui.uri')}}">-->
<!--                    <div class="col-md-4 toplevel-view">-->
<!--                        <h2 class="flaticon-upload14 color-primary-bold">Submit</h2>-->

<!--                        <p>Submit Summary Statistics to GWAS Catalog with our Deposition App</p>-->
<!--                    </div>-->
<!--                </a>-->
<!--                <a th:href="@{/docs/submission}">-->
<!--                    <div class="col-md-4 toplevel-view">-->
<!--                        <h2 class="flaticon-question10 color-primary-bold">How to submit</h2>-->

<!--                        <p>How to Submit Summary Status to GWAS Catalog</p>-->
<!--                    </div>-->
<!--                </a>-->
<!--                <a th:href="@{/docs/summary-statistics-format}">-->
<!--                    <div class="col-md-4 toplevel-view">-->
<!--                        <h2 class="flaticon-notes9 color-primary-bold">Summary statistics format</h2>-->

<!--                        <p>Format for Summary Status Files</p>-->
<!--                    </div>-->
<!--                </a>-->
<!--            </div>-->

            <hr/>
        </div>

    </div>



</div>
<!-- /container -->

<div th:include="fragments/footer :: page_footer"></div>



<!-- Bootstrap core JavaScript
================================================== -->
<!-- Placed at the end of the document so the pages load faster -->
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
<script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.11.2/jquery-ui.min.js"></script>

<script  th:src="@{/js/bootstrap-datepicker.js}"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.1/js/bootstrap.min.js"></script>

<link href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-table/1.11.0/bootstrap-table.css" rel="stylesheet"/>
<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-table/1.11.0/bootstrap-table.js"></script>



<!--bootstrap-table-flat-json-->
<!--<script src="https://cdn.rawgit.com/wenzhixin/bootstrap-table/develop/dist/extensions/flat-json/bootstrap-table-flat-json.js"></script>-->
<!--<script src="https://cdn.rawgit.com/wenzhixin/bootstrap-table/develop/dist/extensions/multiple-sort/bootstrap-table-multiple-sort.js"></script>-->
<!--<script src="https://cdn.rawgit.com/wenzhixin/bootstrap-table/develop/dist/extensions/export/bootstrap-table-export.js"></script>-->
<!--<script src="https://cdn.rawgit.com/wenzhixin/09e218c884f4f380b68a/raw/5d258d2f78225a4ac1a72d54a34cc7976d630df1/jquery.base64.js"></script>-->
<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-table/1.16.0/extensions/export/bootstrap-table-export.js"></script>

<!-- Latest compiled and minified JavaScript -->
<!--<script src="//cdnjs.cloudflare.com/ajax/libs/bootstrap-table/1.5.0/bootstrap-table.min.js"></script>-->
<!-- IE10 viewport hack for Surface/desktop Windows 8 bug -->
<script  th:src="@{/js/ie10-viewport-bug-workaround.js}"></script>

<!-- GOCI JavaScript
================================================== -->
<script th:src="@{/js/goci-ui.js}"></script>
<script th:src="@{/js/helpers/utils-helper.js}"></script>
<script th:src="@{/js/helpers/promise-helper.js}"></script>
<script th:src="@{/js/helpers/utils-helper.js}"></script>
<script th:src="@{/js/common-datatables/summary-stats-datatable.js}"></script>

<!-- this file must be the last. onload action -->
<script  th:src="@{/js/fullpvalueset.js}"></script>
<!-- loading overlay
================================================== -->
<script src="https://cdn.jsdelivr.net/jquery.loadingoverlay/latest/loadingoverlay.min.js"></script>

<!--this come last to make sure trait page specific style is not overwritten by othe css.-->
<link rel="stylesheet" th:href="@{/css/goci-efotrait-page.css}"/>

<!-- Google analytics
================================================== -->
<script>
    (function(i, s, o, g, r, a, m) {
        i['GoogleAnalyticsObject'] = r;
        i[r] = i[r] || function() {
                    (i[r].q = i[r].q || []).push(arguments)
                }, i[r].l = 1 * new Date();
        a = s.createElement(o),
                m = s.getElementsByTagName(o)[0];
        a.async = 1;
        a.src = g;
        m.parentNode.insertBefore(a, m)
    })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
    ga('create', 'UA-60195133-1', 'auto');
    ga('send', 'pageview');
</script>

</body>
</html>
