= Summary statistics

== Summary statistics eligibility

The GWAS Catalog now hosts full summary statistics files (aggregate p-values and association data for every variant analysed in a study) in addition to curated associations.
For summary statistics FAQ, please see https://www.ebi.ac.uk/gwas/docs/faq[here].

Summary statistics must meet a number of criteria to be eligible for inclusion in the GWAS Catalog:

* Must include >100,000 SNPs
* Must reproduce the initial GWAS extracted in the Catalog, i.e. same samples (not a sub-set, e.g. one cohort of a meta-analysis), and on the same trait
* Must be freely available, i.e. not only by request to author or password-protected
* Must not require agreement to any specific terms and conditions, unless this has been agreed in advance by the author/consortium


== Summary statistics acquisition

Summary statistics are acquired from a number of different sources:

* Included in a curated paper as a supplementary file
* Identified via link or reference in a curated paper
* Identified via a consortium or author website
* Submitted to the GWAS Catalog by an author

We encourage authors of GWAS publications included in the Catalog to contact us if summary statistics are available.
If you wish to deposit summary statistics in the GWAS Catalog, please contact us via gwas-subs@ebi.ac.uk for instructions on how to access our FTP server. We can then make these available through the Catalog.
We can only include data in the Catalog that has been published in a peer-reviewed journal and indexed in PubMed. We arenâ€™t currently able to provide accession numbers before publication. If you have an accepted publication for which you intend to submit summary statistics to the GWAS Catalog, and need to include a data availability statement in your manuscript, please contact us.
Note that for summary statistics to be made available through the GWAS Catalog, your study must fulfil our eligibility criteria (see below).

== Summary statistics database

We store summary statistics files in two different places, on our ftp://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/[FTP site], and in harmonised format in our database.

The FTP site hosts the original files and, when they are processed, a subdirectory called 'harmonised' containing a README file, a standard formatted version of the original file (see below) and a harmonised version of the file (also below). The REST API for the summary statistics database is now available at https://www.ebi.ac.uk/gwas/summary-statistics/api/.
For full documentation on the REST API, please see https://www.ebi.ac.uk/gwas/summary-statistics/docs/[here].

== Summary statistics standards and harmonisation

We have proposed a standard format for GWAS summary statistics (Buniello and MacArthur, 2019). To enable users to access data with ease, we provide standard format and harmonised summary statistics files.

The formatted file:

- Is saved as .tsv
- Contains mandatory fields in the following combinations: 

   a) variant_ID and p-value 
   OR 
   b) chromosome, base pair location and p-value
   
Additional standard column headings are listed below.

Standard format file headings

    'variant_id' = variant ID
    'p-value' = p-value
    'chromosome' = chromosome
    'base_pair_location' = base pair location
    'odds_ratio' = odds ratio
    'ci_lower' = lower 95% confidence interval
    'ci_upper' = upper 95% confidence interval
    'beta' = beta
    'standard_error' = standard error
    'effect_allele' = effect allele
    'other_allele' = other allele
    'effect_allele_frequency' = effect allele frequency

Note that the headers in the formatted file are not limited to the above headers.

Formatted data is accessible on the GWAS Catalog dedicated summary statistics link:ftp://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/[FTP site]. Where the formatted files are available, they are nested within the 'harmonised' directory in the study directory.


=== Harmonised summary statistics data

Formatted files are harmonised using an automated link:https://github.com/EBISPOT/sum-stats-formatter/tree/master/harmonisation[pipeline], which is detailed below. By 'harmonised' we mean that all genomic position data will be reported against the same and latest genome build. Missing chromosome, base pair location and variant ID data that can be inferred, will be. Furthermore, where possible, alleles are also orientated to the forward strand.

The harmonisation process is the following:

. Mapping variant IDs to locations
.. Update base pair location value by mapping variant ID using latest Ensembl release, or
.. if above not possible, liftover base pair location to latest genome build, or
.. if above not possible, remove variant from file.

. Orientation (link:https://github.com/opentargets/sumstat_harmoniser[Open Targets project])
.. Using chromosome, base pair location and the effect and other alleles, check the orientation of all non-palindromic variants against Ensembl VCF references to detemine consensus:
* forward
* reverse
* mixed
.. If the consensus is 'forward' or 'reverse', the following harmonisation steps are performed on the palindromic variants, with the assumption that they are orientated according to the consensus, otherwise palindromic variants are not orientated.
.. Using chromosome, base pair location and the effect and other alleles, query each variant against the Ensembl VCF reference to harmonise as appropriate by either:
* keeping record as is because:
** it is already correctly orientated
** it cannot be orientated
* orientating to reference strand:
** reverse complement the effect and other alleles
* flipping the effect and other alleles
** because the effect and other alleles are flipped in the reference
** this also means the beta, odds ratio, 95% CI and effect allele frequency are inverted
* a combination of the orientating and flipping the alleles.
.. The result of the orientation is the addition of a set of new fields for each record (see below). A harmonisation code is assigned to each record indicating the harmonisation process that was performed (note that currently any processes involving 'Infer strand' are not being used).

. Filtering and QC
.. Variant ID is set to variant IDs found by step (5).
.. Records without a valid value for variant ID, chromosome, base pair location and p-value are removed.

==== Table of harmonisation codes

[%header, cols="1*>,10"]
|===
|Code
|Description of process

|1
|Palindromic; Infer strand; Forward strand; Alleles correct

|2
|Palindromic; Infer strand; Forward strand; Flipped alleles

|3
|Palindromic; Infer strand; Reverse strand; Alleles correct

|4
|Palindromic; Infer strand; Reverse strand; Flipped alleles

|5
|Palindromic; Assume forward strand; Alleles correct

|6
|Palindromic; Assume forward strand; Flipped alleles

|7
|Palindromic; Assume reverse strand; Alleles correct

|8
|Palindromic; Assume reverse strand; Flipped alleles

|9
|Palindromic; Drop palindromic; Not orientated

|10
|Forward strand; Alleles correct

|11
|Forward strand; Flipped alleles

|12
|Reverse strand; Alleles correct

|13
|Reverse strand; Flipped alleles

|14
|Required fields are not known; Not orientated

|15
|No matching variants in reference VCF; Not orientated

|16
|Multiple matching variants in reference VCF; Not orientated

|17
|Palindromic; Infer strand; EAF or reference VCF AF not known; Not orientated

|18
|Palindromic; Infer strand; EAF < specified minor allele frequency threshold; Not orientated
|===

- Headers will be coerced to the 'harmonised format'.
- Addition harmonised data columns will be added.
- Rows may be removed.
- Variant ID, chromosome and base pair location may change (likely).


Harmonised file headings (not all may be present in file):

    'variant_id' = variant ID
    'p-value' = p-value
    'chromosome' = chromosome
    'base_pair_location' = base pair location
    'odds_ratio' = odds ratio
    'ci_lower' = lower 95% confidence interval
    'ci_upper' = upper 95% confidence interval
    'beta' = beta
    'standard_error' = standard error
    'effect_allele' = effect allele
    'other_allele' = other allele
    'effect_allele_frequency' = effect allele frequency
    'hm_variant_id' = harmonised variant ID
    'hm_odds_ratio' = harmonised odds ratio
    'hm_ci_lower' = harmonised lower 95% confidence interval
    'hm_ci_upper' =  harmonised lower 95% confidence interval
    'hm_beta' = harmonised beta
    'hm_effect_allele' = harmonised effect allele
    'hm_other_allele' = harmonised other allele
    'hm_effect_allele_frequency' = harmonised effect allele frequency
    'hm_code = harmonisation code (to lookup in 'Harmonisation Code Table')

Headings prefixed with 'hm_' are the output from the orientation (step 2) in the harmonisation process (see above). With the exception of 'hm_code', these will take the value of 'NA' where it was not possible to orientate the allele. In this situation, the data are still harmonised according to definition given above, but the orientation was not possible.

There may be differences between the data available on the FTP site compared to the database where number rounding has taken place. In the database, chromosome values of 'X', 'Y' and 'MT' will appear as 23, 24 and 25, respectively.
