= Summary statistics 

:imagesdir: ../images

This page includes general documentation about the summary statistics we host in the GWAS Catalog. **For instructions on submitting summary statistics please see our https://www.ebi.ac.uk/gwas/docs/submission[submission documentation].**

'''
==== General

- <<What are summary statistics?, What are summary statistics?>>

- <<Sources,Sources>>

- <<Eligibility,Eligibility>>

- <<Standard format,Standard format>>

==== Access

- <<Searching via our web interface:,Searching our web interface>>

- <<Navigating the FTP:,Navigating the FTP>>

- <<Directories:,Directories>>

* <<Harmonised subdirectories:,Harmonised subdirectories>>

- <<How can I find all the harmonised data?,How can I find all the harmonised data?>>

==== Formatted and harmonised files

- <<What are “formatted files”?,What are “formatted files”?>>

- <<What are "harmonised files"?,What are "harmonised files"?>>

* <<Table of harmonisation codes,Table of harmonisation codes>>

==== Accompanying metadata

- <<Accompanying metadata (-meta.yaml),-meta.yaml info>>

* <<File_types:,File types>>

==== API access

- <<REST API,REST API>>

'''
== What are summary statistics?

While associations included within the Catalog are the curated, significant (p<1e-5) and independent loci reported within a paper, summary statistics (sumstats) are the full, aggregate p-values and association data for every variant analysed in a study. These are hosted on our servers when available and can be downloaded via links on study and publication pages.
For more summary statistics FAQ, please see https://www.ebi.ac.uk/gwas/docs/faq#faq-H1[here].

== Sources

Summary statistics are acquired from a number of different sources:

* Submitted to the GWAS Catalog by an author
* Identified via link or reference in a curated paper
* Identified via a consortium or author website
* Included in a curated paper as a supplementary file

We encourage authors to submit summary statistics to the GWAS Catalog. Summary statistics for published and pre-published/unpublished GWAS can be submitted directly via the http://www.ebi.ac.uk/gwas/deposition[Submission page]. Instructions on how to submit summary statistics can be found https://www.ebi.ac.uk/gwas/docs/submission[here].

== Eligibility

Summary statistics must meet a number of criteria to be eligible for inclusion in the GWAS Catalog:

* Must include >100,000 SNPs
* Must reproduce a GWAS described in a published paper (ie. same sample metadata and trait), or is clearly marked as pre/un-published.
* Must be freely available, i.e. not only by request to author or password-protected.
* Must not require agreement to any specific terms and conditions, unless this has been agreed in advance by the author/consortium. Since 2022, we have required newly submitted sumstats to be released under CC0 - noted under ‘Terms/Licence’ on study pages. In rare cases, special licensing terms will appear in an adjacent readme.txt.

== Standard format
The vast majority of files adhere to one of two standard formats, GWAS-SSF and pre-GWAS-SSF. 

Since April 2023, all summary statistics will meet GWAS-SSF - the product of a series of working groups, which has been written up as a https://www.biorxiv.org/content/10.1101/2022.07.15.500230v2[preprint]. The specification of the standard can be found https://github.com/EBISPOT/gwas-summary-statistics-standard/blob/master/gwas-ssf_v1.0.0.pdf[here] with further details in the https://github.com/EBISPOT/gwas-summary-statistics-standard[GitHub repository]. Prior to V1 of this standard, a minimum acceptable format (pre-GWAS-SSF) was used, the details of which are available in https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6323933/[Buniello and MacArthur, 2019].

Standardised data files are additionally provided in a <<What are harmonised files?,harmonised format>>, with all data reported against the same genome build and alleles aligned to the forward strand. Each file has an accompanying <<Accompanying metadata (-meta.yaml),metadata>> file in .yaml format, which indicates the <<File_types:,file_type>> and whether or not the file has been harmonised. 

We also host a small number of non-standard files with eg. fewer than 100,000 SNPs or  non-standard study design (eg. CNV or gene-based). These will be flagged as ‘Non-GWAS-SSF’ in the .yaml.

== Access

We store summary statistics files in two different places, on our https://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/[FTP site], and a subset of files in our database (see <<REST API,REST API>>). The FTP is updated nightly with any newly ingested data.

=== Searching via our web interface: 

image:sumstats_logo.png[Sumstats logo,50]

Links to FTP directories are provided within the main Catalog pages. In this way, summary statistics can be found by searching for a specific trait, accession ID, PMID, author and other key words. Sumstats availability is denoted by the above icon, and listed under the summary statistics tab of a publication. 


=== Navigating the FTP:

Every summary statistics file is available at a path based on the GCST accession ID. For example, the summary statistics for accession GCST123456 would be available at 'ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST123001-GCST124000/GCST123456/'. You can see that the directory 'GCST123456/' is nested inside 'ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST123001-GCST124000/'. This parent directory holds up to 1000 GCSTs, and will only contain those GCSTs that are within the range given. These bins of one thousand are added as necessary, when new GCSTs need housing.

=== Directories:

Each accession ID directory contains:

* The original sumstats file .tsv
* A metadata .yaml
* A separate readme, if provided .txt (newer folders will hold this under ‘author_notes’ inside the .yaml)
* When possible, and once processed, a harmonised directory

=== Harmonised subdirectories:

**The latest version of the harmonisation pipeline was released in 2023.** All files harmonised under this version, eg. https://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST90270001-GCST90271000/GCST90270927/[GCST90270927] contain a ‘/harmonised’ directory with the following contents:

* A harmonised version of the raw file. .h.tsv (see <<What are harmonised files?,What are harmonised files?>>)
* An index file .tbi
* A log file .log
* A metadata file .yaml

**For files harmonised before 2023,** eg https://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST90001001-GCST90002000/GCST90001686/[GCST90001686], the harmonised folder will contain:

* A formatted version of the raw file - with standardised headers .f.tsv (see <<What are “formatted files”?,What are "formatted files"?>>)
* A harmonised version of the formatted file .h.tsv
* A metadata file .yaml

In some cases, there will be no harmonised directory due to incompatibility with our pipeline. These will be marked out with ‘Non-GWAS-SSF’ under file_type in the .yaml. If the file type is GWAS-SSF or pre-GWAS-SSF and there is no harmonised file, it is likely awaiting harmonisation, please check back at a later date.

=== How can I find all the harmonised data?

Harmonised summary statistics are released daily and a text file listing all of the harmonised studies is available on the FTP at the following location: https://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/harmonised_list.txt. This text file is updated on a nightly basis.

== What are “formatted files”?

Prior to the GWAS Catalog accepting direct submissions, summary statistics files were formatted by scientific curators using a semi-automated process. Where the formatted files are available, they are nested within the 'harmonised' subdirectory with the ‘*.f.tsv.gz’ suffix (https://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST90001001-GCST90002000/GCST90001686/harmonised/[example]). These files will not be available for newer submitted content because those datasets have already been formatted by the authors.
During the formatting process data were not manipulated but the headers were converted to a consistent format across all summary statistics files.

== What are "harmonised files"?

Sumstats files are harmonised using an automated link:https://github.com/EBISPOT/gwas-sumstats-harmoniser[pipeline], which is detailed below. By 'harmonised' we mean that all genomic position data will be reported against the same and latest genome build. Missing chromosome, base pair location and variant ID data that can be inferred, will be. Furthermore, where possible, alleles are also orientated to the forward strand.

The harmonisation process is the following:

. Mapping variant IDs to locations (method used is reported in the ‘hm_coordinate_conversion’ field in the harmonised data file)
.. Update base pair location value by mapping rsID using Ensembl (currently v95) (‘hm_coordinate_conversion’ = ‘rs’);  or
.. if above not possible, liftover base pair location to latest genome build (‘hm_coordinate_conversion’ = ‘lo’); or
.. if above not possible, remove variant from file.

. Orientation (link:https://github.com/opentargets/sumstat_harmoniser[Open Targets project])
.. Infer the orientation of palindromic variants. Palindromic variants are alleles on the forward strand which are the same as on the reverse strand. We infer the orientations of palindromic variants according to the strand consensus.
... Firstly, we randomly select 10% of sites. The effect and other alleles are compared with counterpart alternative and reference alleles in the Ensembl VCF references to identify the strand of the non-palindromic variants (forward or reverse) .
... The forward strand consensus can be calculated by forward/(forward+reverse) or reverse/(forward+reverse). To avoid any possibility of sampling bias:
* If the rate  ≥ 0.995,  the following harmonisation steps on the palindromic variants are inferred as on the forward (or reverse) strand;
* If the rate is the range of (0.995,0.9), this rate is recalculated by all non-palindromic variants in the data. The palindromic variants can be inferred as forward (or reverse) if the recalculated rate > 0.99, otherwise palindromic variants are dropped for harmonisation;
* If the rate ≤ 0.9,  palindromic variants are dropped in the following harmonisation step.
.. Variant harmonisation: Using chromosome, base pair location and the effect and other alleles, query each variant against the Ensembl VCF reference to harmonise as appropriate by either:
... keeping record as is because:
* it is already correctly orientated
... orientating to reference strand:
* reverse complement the effect and other alleles
... flipping the effect and other alleles
* because the effect and other alleles are flipped in the reference
* this also means the beta, odds ratio, 95% CI and effect allele frequency are inverted
... a combination of the orientating and flipping the alleles.
... replace with NA because:
* There is no counterpart record in the reference VCF file.

. Filtering and QC
.. Variant ID is set to variant IDs found by step (2).
.. Records without a valid value for variant ID, chromosome, base pair location and p-value are removed.

. Harmonisation result: For each input file, harmonisation generates three outputs:
.. The harmonisation result (*.h.tsv.gz)
* represents harmonised mandatory columns in a https://github.com/EBISPOT/gwas-summary-statistics-standard/blob/master/gwas-ssf_v0.1-draft.pdf[specific order] and other columns in the original order. 
* contains a new column named harmonisation code is assigned to each record indicating the harmonisation process that was performed (note that currently any processes involving 'Infer strand' are not being used).
* is sorted by chr and position and compressed using bgzip
.. An index file (*.h.tsv.gz.tbi) 
* A tabix index file of the harmonisation result for quick data retrieve purposes
.. A report file (report.txt) summarises:
* the reference VCF file used in harmonisation.
* process on the palindromic variants
* percentage of variants dropped in the qc step
* percentage of variants that are successfully harmonised or failed
.. A metadata YAML file (*.h.tsv.gz-meta.yaml) for the harmonised data file. 

===== Table of harmonisation codes

[%header, cols="1*>,10"]
|===
|Code
|Description of process

|1
|Palindromic; Infer strand; Forward strand; Alleles correct

|2
|Palindromic; Infer strand; Forward strand; Flipped alleles

|3
|Palindromic; Infer strand; Reverse strand; Alleles correct

|4
|Palindromic; Infer strand; Reverse strand; Flipped alleles

|5
|Palindromic; Assume forward strand; Alleles correct

|6
|Palindromic; Assume forward strand; Flipped alleles

|7
|Palindromic; Assume reverse strand; Alleles correct

|8
|Palindromic; Assume reverse strand; Flipped alleles

|9
|Palindromic; Drop palindromic; Not orientated

|10
|Forward strand; Alleles correct

|11
|Forward strand; Flipped alleles

|12
|Reverse strand; Alleles correct

|13
|Reverse strand; Flipped alleles

|14
|Required fields are not known; Not orientated

|15
|No matching variants in reference VCF; Not orientated

|16
|Multiple matching variants in reference VCF; Not orientated

|17
|Palindromic; Infer strand; EAF or reference VCF AF not known; Not orientated

|18
|Palindromic; Infer strand; EAF < specified minor allele frequency threshold; Not orientated
|===

Harmonised data files conform to the standard but there are some additional points to mention:

. Where possible, data are sorted by chromosome and base pair location
. Where possible, a <data_file>.tbi file is available to enable access to a specified genomic region using tabix (link)
. A running report from the harmonisation is available with the harmonised file on the FTP
. A field in the data file, ‘‘hm_coordinate_conversion’’, signifies whether the position was mapped using rsID (‘rs’) lookup or liftOver (‘lo’).

There may be differences between the data available on the FTP site compared to the database where number rounding has taken place. In the database, chromosome values of 'X', 'Y' and 'MT' will appear as 23, 24 and 25, respectively.

== Accompanying metadata (-meta.yaml)
Metadata are provided in a computer and human readable YAML file adjacent to the data file, suffixed with -meta.yaml. So where the data file is called GCST123456.tsv.gz, the metadata can be found with the name GCST123456.tsv.gz-meta.yaml. 

*Metadata definitions:*
[options="header"]
|=============================================================================================================================================================
|  *Field*                   |  *Description*                  |  *Data type and values*                  |  *Mandatory*                    |  *Example*                                                                                                              
|--------------------------|----------------------------------|------------------|----------------|-------------------------------------
|  # *Study meta-data* |                                                                                      |                                                           |                                       |                                                                                                                       
|  gwas_id                           |  GWAS Catalog accession ID                                                           |  Text string                                              |  Yes                                  |  GCST90244057                                                                                                         
|  author_notes                      |  Additional information about this study from the author                             |  Text string                                              |  No                                   |  File contains GWAS summary statistics from a meta-analysis of NMR metabolic traits in up to 33 cohorts.              
|  gwas_catalog_api                  |  GWAS catalog REST API link                                                          |  Text string                                              |  Yes                                  |  https://www.ebi.ac.uk/gwas/rest/api/studies/GCST90244057
|  date_metadata_last_modified       |  The latest date that metadata YAML file was modified                                |  date                                                     |  Yes                                  |  2023-11-28                                                                                                           
|  # *Trait Information*              |                                                                                      |                                                           |                                       |                                                                                                                       
|  trait_description                 |  Author reported trait description                                                   |  Text string (multiple possible)                          |  Yes                                  |  Body mass index                                                                                                      
|  ontology_mapping                  |  Short form ontology terms describing the trait                                      |  Text string (multiple possible)                          |  No                                   |   EFO_0004918                                                                                                         
|  # *Genotyping Information*         |                                                                                      |                                                           |                                       |                                                                                                                       
|  genome_assembly                   |  Genome assembly for the summary statistics.                                         |  GRCh/NCBI/UCSC value                                     |  Yes                                  |  GRCh37                                                                                                               
|  coordinate_system                 |  Coordinate system used for the summary statistics                                   |  Text String (1-based or 0-based)                         |  No                                   |  1-based                                                                                                              
|  genotyping_technology             |  Method(s) used to genotype variants in the discovery stage.                         |  Text string (multiple possible)                          |  Yes                                  |   Genome-wide genotyping array                                                                                        
|  imputation_panel                  |  Panel used for imputation                                                           |  Text string                                              |  No                                   |  HRC + UK10K                                                                                                          
|  imputation_software               |  Software used for imputation                                                        |  Text string                                              |  No                                   |  SHAPEIT3 + IMPUTE4                                                                                                   
|  # *Sample Information*             |                                                                                      |                                                           |                                       |                                                                                                                       
|  sample_ancestry_category          |  Broad ancestry category that best describes the sample.                             |  Text string                                              |  Yes                                  |  European                                                                                                             
|  sample_ancestry                   |  The most detailed ancestry descriptor(s) for the sample.                            |  Text string (multiple possible)                          |  Yes                                  |  Finnish                                                                                              
|  sample_size                       |  Sample size                                                                         |  Integer                                                  |  Yes                                  | 27006                                                                                                                 
|  ancestry_method                   |  Method used to determine sample ancestry e.g. self-reported/genetically determined  |  Text string (multiple possible)                          |  No                                   |  self-reported                                                                                                        
|  case_control_study                |  Flag whether the study is a case-control study                                      |  Boolean                                                  |  No (default is false)                |  true                                                                                                                 
|  case_count                        |  Number of cases for case/control study                                              |  Integer                                                  |  No, unless caseControlStudy is true  | 27006                                                                                                                 
|  control_count                     |  Number of controls for case/control study                                           |  Integer                                                  |  No, unless caseControlStudy is true  | 27006                                                                                                                 
|  sex                               |  To indicate a sex-stratified analysis                                               |  M (for male), F (for female), combined or NR if unknown  |  No                                   |  combined                                                                                                             
|  # *Summary Statistic information*  |                                                                                      |                                                           |                                       |                                                                                                                       
|  data_file_name                    |  The name of the summary statistics file                                             |  Text string                                              |  Yes                                  |  GCST90244057_buildGRCh37.tsv                                                                                         
|  file_type                         |  The format of the summary statistics file                                           |  "GWAS-SSF v1.0", "pre-GWAS-SSF", "non-GWAS-SSF"          |  Yes                                  |  GWAS-SSF v1.0                                                                                                        
|  data_file_md5sum                  |  The md5 checksum of the summary statistics file.                                    |  Text string                                              |  Yes                                  |  0ec56396f89edcc21a3d5a25a6fa993d                                                                                     
|  analysis_software                 |  Software and version used for the association analysis                              |  Text string (multiple possible)                          |  Yes if p-values of 0 given           |  REGENIE                                                                                                              
|  adjusted_covariates               |  Any covariates the GWAS is adjusted for                                             |  Text string (multiple possible)                          |  No                                   |   sex                                                                                                                 
|  minor_allele_freq_lower_limit     |  Lowest possible effect allele frequency                                             |  Numeric                                                  |  No                                   | 0.0003                                                                                                                
|  # *Harmonization status*           |                                                                                      |                                                           |                                       |                                                                                                                       
|  is_harmonised                     |  Description of harmonisation codes                                                  |  Text string                                              |  Only given in harmonised datasets    |  false                                                                                                                
|  is_sorted                         |  Flag whether the file is sorted by genomic location                                 |  Boolean                                                  |  Yes                                  |  false                                                                                                                
|  harmonisation_reference           |  The genome reference file used for harmonising the summary statistics file          |  Text string                                              |  No                                   |   	ftp://ftp.ensembl.org/pub/release-104/fasta/homo_sapiens/dna/                                                                                                                     
|
|=============================================================================================================================================================

==== File_types:

* GWAS-SSF v1: Version 1 of the GWAS-SSF standard format.  
* Pre-GWAS-SSF: The previous, minimal format requiring only chromosome, base_pair_location/rsID and p-value. 
* Non-GWAS-SSF: Not conforming to any standard format. Can include gene-based, CNV and multivariate analyses where eg. effect information is not available.

Note: for very old sumstats (where the filename does not begin ‘GCST’), no metadata .yaml will be available.

For summary statistics associated with a journal publication, study metadata can also be accessed via the GWAS Catalog web interface and REST API. 
Metadata for unpublished submissions is not currently available via REST API but can be found in our https://www.ebi.ac.uk/gwas/docs/file-downloads[download files].

== REST API
The REST API for the summary statistics database is available at https://www.ebi.ac.uk/gwas/summary-statistics/api/. For full documentation on the REST API, please see https://www.ebi.ac.uk/gwas/summary-statistics/docs/[here].
It’s important to note that the data available via REST API is a subset of that available via FTP. Currently, we’re unable to release more data to the REST API as it’s undergoing a complete redevelopment to help us cope with the tremendous growth in summary statistics data.
